src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,210 @@
+            this.next = next;
+        }
+
+        public final K getKey() {
+            return key;
+        }
+
+        public final V getValue() {
+            return value;
+        }
+
+        public final V setValue(V newValue) {
+            V oldValue = value;
+            value = newValue;
+            return oldValue;
+        }
+
+        public final int hashCode() {
+            hash = 31;
+            hash = hash * 17 + key.hashCode();
+            return hash;
+        }
+
+        @Override
+        public String toString() {
+            return "Node{" + "hash=" + hash + ", key=" + key
+                    + ", value=" + value + ", next=" + next + '}';
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V>[] tab;
+        Node<K,V> p;
+        int n;
+        if ((tab = table) == null || (n = tab.length) == 0) {
+            n = (tab = resize()).length;
+        }
+        int hashValue = hash(key);
+        int idx = indexFor(hashValue,n);
+        if ((p = tab[idx]) == null) {
+            tab[idx] = newNode(hashValue, key, value, null);
+        } else {
+            Node<K,V> e;
+            K k;
+            if (p.hash == hashValue && ((k = p.key) == key
+                    || (key != null && key.equals(k)))) {
+                e = p;
+            } else {
+                for (int in = 0; ; ++in) {
+                    if ((e = p.next) == null) {
+                        p.next = newNode(hashValue, key, value, null);
+                        break;
+                    }
+                    if (e.hash == hashValue && ((k = e.key) == key
+                            || (key != null && key.equals(k)))) {
+                        break;
+                    }
+                    p = e;
+                }
+            }
+            if (e != null) {
+                e.value = value;
+                return;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(hash(key), key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V>[] resize() {
+        Node<K, V>[] oldTab = table;
+        int oldCap = (oldTab == null) ? 0 : oldTab.length;
+        int oldThr = threshold;
+        int newCap;
+        int newThr;
+        if (oldCap > 0) {
+            newCap = oldCap << 1;
+            newThr = oldThr << 1;
+        } else {
+            newCap = INIT_SIZE;
+            newThr = (int)(INIT_LOAD_FACTOR * INIT_SIZE);
+        }
+        threshold = newThr;
+        Node<K,V>[] newTab = (Node<K,V>[]) new Node[newCap];
+        table = newTab;
+        if (oldTab != null) {
+            transfer(oldTab,newTab);
+        }
+        return newTab;
+    }
+
+    private void transfer(Node<K,V>[] oldTab, Node<K,V>[] newTab) {
```

Comment:
This whole method is absolutely massive and has terrible readability. 
First of all, have a proper naming for ALL the variables and not just 1 letter.
I am sure we can optimize it or at least find some common parts with `put()` method and split it into several private methods.